docs/DEVELOPMENT.md

```diff
@@ -2,6 +2,85 @@
 # Developer's guide to Qdrant
 
 
+## Build Qdrant
+
+### Docker 🐳
+
+Build your own from source
+
+```bash
+docker build . --tag=qdrant/qdrant
+```
+
+Or use latest pre-built image from [DockerHub](https://hub.docker.com/r/qdrant/qdrant)
+
+```bash
+docker pull qdrant/qdrant
+```
+
+To run the container, use the command:
+
+```bash
+docker run -p 6333:6333 qdrant/qdrant
+```
+
+And once you need a fine-grained setup, you can also define a storage path and custom configuration:
+
+```bash
+docker run -p 6333:6333 \
+    -v $(pwd)/path/to/data:/qdrant/storage \
+    -v $(pwd)/path/to/custom_config.yaml:/qdrant/config/production.yaml \
+    qdrant/qdrant
+```
+
+* `/qdrant/storage` - is a place where Qdrant persists all your data. 
+Make sure to mount it as a volume, otherwise docker will drop it with the container. 
+* `/qdrant/config/production.yaml` - is the file with engine configuration. You can override any value from the [reference config](https://github.com/qdrant/qdrant/blob/master/config/config.yaml) 
+
+Now Qdrant should be accessible at [localhost:6333](http://localhost:6333/).
+
+
+### Local development
+#### Linux/Debian
+To run Qdrant on local development environment you need to install below:
+- Install Rust, follow: [install rust](https://www.rust-lang.org/tools/install)
+- Install `rustfmt` toolchain for Rust 
+    ```shell
+    rustup component add rustfmt
+    ```
+- Install dependencies:
+    ```shell
+    sudo apt-get update -y 
+    sudo apt-get upgrade -y
+    sudo apt-get install -y curl unzip gcc-multilib \
+        clang cmake jq \
+        g++-9-aarch64-linux-gnu \
+        gcc-9-aarch64-linux-gnu
+    ```
+- Install `protoc` from source
+    ```shell
+    PROTOC_VERSION=22.2
+
+    # curl `proto` source file 
+    curl -LO https://github.com/protocolbuffers/protobuf/releases//download/v$PROTOC_VERSION/protoc-$PROTOC_VERSION-linux-x86_64.zip
+
+    unzip protoc-$PROTOC_VERSION-linux-x86_64.zip -d $HOME/.local
+
+    export PATH="$PATH:$HOME/.local/bin"
+
+    # remove source file if not needed
+    rm protoc-protoc-$PROTOC_VERSION-linux-x86_64.zip
+    
+    # check insalled `protoc` version
+    protoc --version
```

Comment:
Could you please add a comment about what version do you expect to see here?

My local setup says 

```
protoc --version
libprotoc 3.12.4
```

which might be confusing, as we were installing `PROTOC_VERSION=22.2`

Reply:
I'm not sure which minimum version, but I found that version of `protoc` which installed by `apt` as below is not meet the requirements of Qdrant.
Check Issue https://github.com/qdrant/qdrant/issues/1385.

For latest `protoc` version by `apt` is: 3.12.4
```
$ sudo apt install protobuf-compiler
...
$ protoc --version
libprotoc 3.12.4
```

But when installing it from source with latest release you will get below:
```
$ protoc --version
libprotoc 22.2
```

Reply:
Also I didn't test it on `docker` yet, but I'm confused how it is working on `docker`!

Reply:
It looks like debian and ubuntu have different versions pinned in APT:

https://packages.ubuntu.com/search?suite=default&section=all&arch=any&keywords=protobuf-compiler&searchon=names
https://packages.debian.org/sid/protobuf-compiler

Reply:
But also I am confused why version semantic is so different - `22.2` vs `3.12.4`. Is it different forks or what is the reason?

Reply:
Regarding versioning, I did investigate this and did read many resources but I have some confusion. 
Generally it is common in the Protocol Buffers.

- Official documentation said:
> Versioning
> 
> We changed our versioning scheme to enable more-nimble updates to language-specific parts of Protocol Buffers. In the new scheme, each language has its own major version that can be incremented independently of other languages, as covered later in this topic with the Python release. The minor and patch versions, however, will remain coupled. This allows us to introduce breaking changes into some languages without requiring a bump of the major version in languages that do not experience a breaking change.
> 
> The first instance of this new versioning scheme is the new version of the Python API, 4.21.0, which follows the preceding version, 3.20.1. Other language APIs will be released as 3.21.0

Link: https://protobuf.dev/news/2022-05-06/#versioning

- This is one of the issues related to this topic: https://github.com/protocolbuffers/protobuf/issues/11440

- Below is answer of BingAI:

> Question 1: Why versioning in Protocol Buffers is not linear. For example v3.20.2 and v21.6?
> > Protocol Buffers is a serialization format for packets of typed, structured data that are up to a few megabytes in size. The format is suitable for both ephemeral network traffic and long-term data storage. Protocol Buffers can be extended with new information without invalidating existing data or requiring code to be updated⁵. 
> > Google designed protobuf to be pretty forgiving with versioning: unexpected data is either stored as "extensions" (making it round-trip safe), or silently dropped, depending on the implementation. New fields are generally added as "optional", meaning that old data can be loaded successfully⁴. 
> > The versioning scheme of Protocol Buffers has changed to enable more-nimble updates to language-specific parts of Protocol Buffers. In the new scheme, each language has its own major version that can be incremented independently of other languages⁶.

> > Source: Conversation with Bing, 3/21/2023(1) Overview | Protocol Buffers Documentation - Google Developers. https://protobuf.dev/overview/ Accessed 3/21/2023.
> > (2) How does protocol buffer handle versioning? - Stack Overflow. https://stackoverflow.com/questions/8519381/how-does-protocol-buffer-handle-versioning Accessed 3/21/2023.
> > (3) Changes made on May 6, 2022 | Protocol Buffers Documentation. https://protobuf.dev/news/2022-05-06/ Accessed 3/21/2023.
> > (4) How does protocol buffer handle versioning? - Stack Overflow. https://stackoverflow.com/questions/8519381/how-does-protocol-buffer-handle-versioning Accessed 3/21/2023.
> > (5) protocol buffers - gRPC / Protobuf interface versioning - Stack Overflow. https://stackoverflow.com/questions/40519610/grpc-protobuf-interface-versioning Accessed 3/21/2023.
> > (6) Protobuf: Explanation, Benefits + Tutorial - IONOS. https://www.ionos.com/digitalguide/websites/web-development/protocol-buffers-explained/ Accessed 3/21/2023.

> Question 2: What do you mean enable more-nimble updates to language-specific parts of Protocol Buffers?
> > The versioning scheme of Protocol Buffers has changed to enable more-nimble updates to language-specific parts of Protocol Buffers. In the new scheme, each language has its own major version that can be incremented independently of other languages. This means that changes to one language's implementation of Protocol Buffers will not affect other languages' implementations.